.github/workflows/spellcheck.yml

```diff
@@ -0,0 +1,72 @@
+name: spell_check
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+    - 'main'
+    - 'develop'
+    paths:
+      - 'README.md'
+      - 'CONTRIBUTING.md'
+      - 'notebooks/**.md'
+      - 'notebooks/**.ipynb'
+  push:
+    branches:
+    - 'main'
+    paths:
+      - 'README.md'
+      - 'CONTRIBUTING.md'
+      - 'notebooks/**.md'
+      - 'notebooks/**.ipynb'
+    
+
+jobs:
+  build_spellcheck:
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-20.04  # change cachepip step when changing this
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Dotenv Action
+      id: dotenv
+      uses: xom9ikk/dotenv@v1.0.2
+      with:
+        path: ./.github/workflows
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8  # change cachepip step when changing this
+    - name: Cache OpenVINO Pip Packages
+      id: cachepip
+      uses: actions/cache@v2
+      with:
+        path: |
+          pipcache
+        key: ${{ env.PIP_CACHE_KEY }}-ubuntu-20.04-3.8
+    - name: Cache openvino packages
+      if: steps.cachepip.outputs.cache-hit != 'true'
+      run: |
+        python -m pip install --upgrade pip
+        mkdir pipcache
+        python -m pip install --cache-dir pipcache --no-deps openvino openvino-dev nncf
+        cp -r pipcache pipcache_openvino
+        python -m pip uninstall -y openvino openvino-dev nncf
```

Comment:
Do we need to install OpenVINO for spellcheck?

Reply:
Removed unnecessary stages from GitHub Actions Workflow